src/components/todo/index.tsx

```diff
@@ -1,59 +1,62 @@
 'use client';
 
-import React, { useContext } from 'react';
-import { TodoDataContext } from '@/context/TodoDataContext';
-import { TodoUIContext } from '@/context/TodoUIContext';
+import React, { useEffect } from 'react';
+import { useAppDispatch } from '@/lib/hooks';
+import { setDisplayingDate, setLoading, setTodoListData, setSessionId } from '@/lib/todos/todoDataSlice';
 import TodoHeader from '@/components/todo/TodoHeader';
 import TodoPagination from '@/components/todo/TodoPagination';
 import TodoCalendar from '@/components/todo/TodoCalendar';
 import TodoReport from '@/components/todo/TodoReport';
 import TodoList from '@/components/todo/TodoList';
+import * as S from '@/components/todo/Todo.styled';
+import { useSession } from 'next-auth/react';
+import { useParams } from 'next/navigation';
+import { useGetTodosMatchingDate } from '@/api/hooks/todoHooks';
+import { useMemo } from 'react';
 import TodoModal from './TodoModal';
-import * as S from './Todo.styled';
 
 export default function Todo() {
-  const {
-    // 데이터
-    displayingDate,
-    todoId,
-    setTodoId,
-  } = useContext(TodoDataContext);
-  const {
-    // 달력
-    isCalendarOpen,
-    toggleIsCalendarOpen,
-    // 모달
-    mode,
-    setTodoModalMode,
-    isModalOpen,
-    setIsModalOpenTrue,
-    setIsModalOpenFalse,
-    isModalOpenedByFAB,
-    setIsModalOpenedByFABTrue,
-    setIsModalOpenedByFABFalse,
-  } = useContext(TodoUIContext);
+  const dispatch = useAppDispatch();
+  const { data: session } = useSession();
+  const sessionId = session?.user?.id;
+  // TodoPagination, TodoCalendar에서 선택한 날짜를 가져옴
+  const params = useParams();
+  const { year, month, day } = params;
+
+  const displayingDate = useMemo(() => {
+    return year && month && day ? new Date(Number(year), Number(month) - 1, Number(day)) : new Date();
+  }, [year, month, day]);
+
+  const { data: todoListData = [], isLoading } = useGetTodosMatchingDate(sessionId, displayingDate);
```

Comment:
p3(그러나, api 수정 이후):
```suggestion
  const { data: todoListData , isLoading } = useGetTodosMatchingDate(sessionId, displayingDate);
```
이렇게 수정하여 사용하고 싶지만, 현재 `useGetTodosMatchingDate`가 초기버전이여서 todos가 없으면 에러가 뜨네요 그래서 빈 배열로 초기값을 준 거 같은데, 이후에는 값이 없다면 빈 배열이 오는 방향으로 수정된다고 하니, 해당 부분도 api수정과 같이 바꿔주면 코드 가독성이 더 높아질 것 같아욤

Reply:
오호 좋은 정보 감사합니다! 다음 PR에 반영해보갔습니닷